packages/plugin-solana/src/actions/swap.ts

```diff
@@ -145,12 +144,10 @@ Respond with a JSON markdown block containing only the extracted values. Use nul
 
 // get all the tokens in the wallet using the wallet provider
 async function getTokensInWallet(runtime: IAgentRuntime) {
+    const publicKey = await getPublicKey(runtime);
```

Comment:
`getPublicKey` is not defined. It is also not in the `keypairUtils.ts` file.

Reply:
Do you mean to call `getWalletKey()` to get the publicKey ?

Reply:
Great catch! I refactored the solution and must have missed this. Odd that this passed the checks during `pnpm build`. Thank you for your help!